### PR TITLE
fix: adding one click signup

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/5850

- didn't added extra field support for one click

required https://github.com/fossasia/open-event-server/pull/8076

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by MeghalBisht** [comment]
Works well, needs some changes
1. When not logged in, unauthorized error in console. 
Screenshot from 2021-07-06 15-14-47
2.
> On the event page provide a registration button instead of ticket options with a number drop-down.

**Comment by maze-runnar** [comment]
> On the event page provide a registration button instead of ticket options with a number drop-down.

Is it - On the event page provide a registration button. "instead of ticket options with a number drop-down." **_or_**
On the event page provide a registration button with number drop down , instead of ticket options. 😂

**Comment by mariobehling** [comment]
As discussed in the meeting:
1. You cannot add the registration option for a ticketed event. Please delete the tickets in order to use the registration option.
2. For events that have a registration: You cannot add tickets for events with a registration. Please delete the registration for your event in order to add tickets.

Keep in mind the minimum/max order for registrations is always 1.

**Comment by mariobehling** [comment]
Ok, here is a better idea after thinking more about it.

Screenshot from 2021-07-08 17-57-52

We can add the options on the top of the box "Ticketed Event <> Registration"

Then with the ticketed event option just show things as they are now.

With the registration option show:
* Free registration
* Paid registration
* Donation registration

With these options only show min/max 1. Only 1 registration is possible.

On the event page show a box like that

Screenshot from 2021-07-08 18-05-58

When the user clicks the box direct the user immediately to the next order page. We will work on the specifics of the order page in the next step.

**Comment by maze-runnar** [comment]
> Ok, here is a better idea after thinking more about it.
> 
> Screenshot from 2021-07-08 17-57-52
> 
> We can add the options on the top of the box "Ticketed Event <> Registration"
> 
> Then with the ticketed event option just show things as they are now.
> 
> With the registration option show:
> * Free registration
> * Paid registration
> * Donation registration
> 
> With these options only show min/max 1. Only 1 registration is possible.
> 
> On the event page show a box like that
> 
> Screenshot from 2021-07-08 18-05-58
> 
> When the user clicks the box direct the user immediately to the next order page. We will work on the specifics of the order page in the next step.
> 

Ok

**Comment by MeghalBisht** [comment]
Screenshot from 2021-07-30 23-32-00
Empty registration ticket should not be allowed @maze-runnar

File: app/controllers/public/index.js [hunk 1]
```diff
@@ -122,6 +122,35 @@ export default class IndexController extends Controller {
...
+  async oneClickSignup(ticket) {
+    const input = {
+      tickets: [{
+        id       : ticket.id,
+        quantity : 1,
+        price    : 0
+      }]
+    };
+
+    let myinput = {};
```
Comment (MeghalBisht) [inline app/controllers/public/index.js:135]:
```suggestion
    let myInput = {};
```

File: app/controllers/public/index.js [hunk 2]
```diff
@@ -122,6 +122,35 @@ export default class IndexController extends Controller {
...
+    for (const t of input.tickets) {
+      if (t) {
+        myinput = {
```
Comment (MeghalBisht) [inline app/controllers/public/index.js:138]:
```suggestion
        myInput = {
```

File: app/controllers/public/index.js [hunk 3]
```diff
@@ -122,6 +122,35 @@ export default class IndexController extends Controller {
...
+          tickets: [t]
+        };
+      }
+    }
+    try {
+      const order = await this.loader.post('/orders/create-order', myinput);
```
Comment (MeghalBisht) [inline app/controllers/public/index.js:144]:
```suggestion
      const order = await this.loader.post('/orders/create-order', myInput);
```

**Comment by MeghalBisht** [comment]
1. I added a normal ticket then I switch to registration ticket -> I am able to save without adding a registration ticket.
2. Registration and normal tickets cannot have same name. Is that intended? Eg - If I create a normal ticket named "FREE" and want to create a normal ticket named "FREE". It is not allowed

**Comment by maze-runnar** [comment]
> 1. I added a normal ticket then I switch to a registration ticket -> I am able to save without adding a registration ticket.

Actually, it is checking for complete ticket numbers. 

> 2. Registration and normal tickets cannot have same name. Is that intended? Eg - If I create a normal ticket named "FREE" and want to create a normal ticket named "FREE". It is not allowed

This was known.

**Comment by MeghalBisht** [comment]
@maze-runnar Please remove all the commented code and lets merge it to test more.

**Comment by maze-runnar** [comment]
> @maze-runnar Please remove all the commented code and lets merge it to test more.

@MeghalBisht done!

**Review by MeghalBisht — DISMISSED** [review]

**Review by ranjitkshah — APPROVED** [review]

**Pull request merged**